app/(gcforms)/[locale]/(form filler)/id/[...props]/lib/notifications.ts

```diff
@@ -0,0 +1,246 @@
+import { sendEmail } from "@lib/integration/notifyConnector";
+import { logMessage } from "@lib/logger";
+import { getRedisInstance } from "@lib/integration/redisConnector";
+import { getOrigin } from "@lib/origin";
+import { NotificationsInterval } from "@gcforms/types";
+import { serverTranslation } from "@i18n";
+import { prisma, prismaErrors } from "@lib/integration/prismaConnector";
+import { logEvent } from "@lib/auditLogs";
+
+export const Status = {
+  SINGLE_EMAIL_SENT: "SINGLE_EMAIL_SENT",
+  MULTIPLE_EMAIL_SENT: "MULTIPLE_EMAIL_SENT",
+} as const;
+export type Status = (typeof Status)[keyof typeof Status];
+
+// Sends an email notification when a user has new form submissions
+export const sendNotification = async (formId: string, titleEn: string, titleFr: string) => {
+  const usersAndNotifications = await getUsersAndNotificationsInterval(formId);
+  if (!usersAndNotifications) {
+    logMessage.error(`sendNotification template not found with id ${formId}`);
+    return;
+  }
+
+  const { users } = usersAndNotifications;
+
+  const notificationsInterval =
```

Comment:
To get around explicitly typecasting NotificationsInterval when it could be undefined or Null you can change the type of `validateNotificationInterval` to be a type predicate: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates